src/Equinor.ProCoSys.IPO.Command/InvitationCommands/FillRfocStatuses/FillRfocStatusesCommandHandler.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.McPkg;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.InvitationCommands.FillRfocStatuses
+{
+    public class FillRfocStatusesCommandHandler : IRequestHandler<FillRfocStatusesCommand, Result<Unit>>
+    {
+        private readonly IInvitationRepository _invitationRepository;
+        private readonly IProjectRepository _projectRepository;
+        private readonly IMcPkgApiService _mcPkgApiService;
+        private readonly ICommPkgApiService _commPkgApiService;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly ILogger<FillRfocStatusesCommandHandler> _logger;
+
+        public FillRfocStatusesCommandHandler(
+            IInvitationRepository invitationRepository,
+            IUnitOfWork unitOfWork,
+            ILogger<FillRfocStatusesCommandHandler> logger,
+            IProjectRepository projectRepository,
+            IMcPkgApiService mcPkgApiService,
+            ICommPkgApiService commPkgApiService)
+        {
+            _invitationRepository = invitationRepository;
+            _logger = logger;
+            _unitOfWork = unitOfWork;
+            _projectRepository = projectRepository;
+            _mcPkgApiService = mcPkgApiService;
+            _commPkgApiService = commPkgApiService;
+        }
+
+        public async Task<Result<Unit>> Handle(FillRfocStatusesCommand request, CancellationToken cancellationToken)
+        {
+            // THIS CODE WAS WRITTEN TO RUN A ONETIME TRANSFORMATION WHEN WE INTRODUCED RfocAccepted on McPkgs and CommPkgs,
+            // and new status (ScopeHandedOver) on invitation
+            // WE KEEP THE CODE ... MAYBE WE WANT TO DO SIMILAR STUFF LATER
+
+            var allProjects = await _projectRepository.GetAllAsync();
+            var invitations = await _invitationRepository.GetAllAsync();
```

Comment:
The invitation repo include all type of "child" nodes, which mean that this will be a big select. Is that ok in prod consider the size, or should you create a special repo-method getting invitations with only needed  "child" nodes (McPkgs + CommPkgs)

Reply:
Added method to get invitations with only scope in includes.